uwblueprint/social-diversity-for-children
# Finalized Parent Enrollment

### Notion ticket link

[Finalized Parent Enrollment](https://www.notion.so/uwblueprintexecs/Finalize-Parent-Enrollment-269c8281ab4d4fa58036d3128b153f6f)

### Implementation description

- Connected all the components of the parent class enrollment
- Created discount applied page according to the design on Figma
- Added API call on last page to actually create the registration record



### Steps to test

1. Log in, ensure that the user you log in as has a child that can register for one of the courses in your test data.
2. Try to register for a course through one of the program cards, if that doesn't work go to `parent/enrollment?classId={VALID_CLASS_ID}`
3. Go through all the pages
4. Try all the options for the proof on income page as follows.
5. On the user object, if `isLowIncome = false` and there is no proof of income, then it should lead you to the proof of income page
6. Manually make your user `isLowIncome = true`, this should bring you to the discount page
7. Put a random string into the `proofOfIncome` field of the user , this should skip the proof of income page
8. Continue to the next page, then check the Db to ensure that there was a new registration record created

### What should reviewers focus on?

- There are a few things that I don't love in my implementation, that I could not resolve myself, I will point out here to see if anyone else can figure them out.
    -  In `ProofOfIncomeWrapper.tsx` the logic for `isLowIncome = false` and there is no proof of income should be to skip this page, thus the nextPage() function is called, however this needs to be guarded by the page being on the correct page number or else the rendering will cause the flow to skip to the end. Thus the page number of the income page is hard-coded, and I am not sure of a way to work around it. (line 94 of the file)
    - Once the user uploads their income proof if applicable, they are redirected to the beginning of the flow, I am not sure if that is an acceptable flow, or how to reasonable mitigate it (It might be possible by saving a very specialized value in state and checking when rendering this flow but I am not sure it's worth it)
    - The last page seems to make the API call multiple times when the page is left idle, I am not sure why this happens and how to circumvent it, but since there is the unique guard on the DB, it functions as expected.

### Checklist



### Demo

https://user-images.githubusercontent.com/53587708/138224159-a9697b05-91bb-44a8-9d4a-db49b444a020.mp4
